### PR TITLE
fix integer overflow in Attention

### PR DESCRIPTION
**Pull request by yufenglee**
### Description

offset used in attention is with data type int. It can overflow for large sequence length.

### CODE REVIEW

**Comment by tianleiwu** [comment]
Add a test case of large seq len?

File: onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h [hunk 1]
```diff
@@ -208,7 +208,7 @@ class GQAAttentionBase : public AttentionBase {
                                bool packed_qkv,                     // whether Q, K, V are packed
                                ThreadPool* tp) const {
     const bool is_prompt = sequence_length != 1;
-    const int packed_batch_stride = packed_qkv ? (num_heads_ + 2 * kv_num_heads_) * sequence_length * head_size : 0;
+    const int64_t packed_batch_stride = packed_qkv ? SafeInt<int64_t>(num_heads_ + 2 * kv_num_heads_) * sequence_length * head_size : 0;
```
Comment (kunal-vaishnavi) [inline onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h:211]: Why do the types in these two lines differ?

https://github.com/microsoft/onnxruntime/blob/b65ad66812ee12a5235c197f465db96f25561e0f/onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h#L101
https://github.com/microsoft/onnxruntime/blob/b65ad66812ee12a5235c197f465db96f25561e0f/onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h#L211

**Comment by snnn** [comment]
Actually we have a lot of such issues.  VC++ static analyzer can find most of them. See this: https://github.com/microsoft/onnxruntime/security/code-scanning?query=is%3Aopen+branch%3Amain+tool%3APrefast+language%3Ac%2B%2B+ 
We know the problem, just we didn't have enough time to fix them.

File: onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h (hunk 1 above)
Comment (tianleiwu) [reply onnxruntime/contrib_ops/cpu/bert/gqa_attention_base.h:211]: ptrdiff_t are 32-bit on the 32-bit platform and 64-bit on the 64-bit platform. 32 bit platform will not able to run large seq length  anyway.

**Review by kunal-vaishnavi — APPROVED** [review]

**Review by tianleiwu — APPROVED** [review]

**Review by aciddelgado — APPROVED** [review]

**Pull request merged**